clients/cli/src/prover.rs

```diff
@@ -7,132 +7,251 @@ use crate::orchestrator_client::OrchestratorClient;
 use crate::setup;
 use crate::utils;
 use colored::Colorize;
+use crossterm::{
+    cursor::{Hide, MoveTo, Show},
+    execute,
+    style::{Color, Print, ResetColor, SetForegroundColor},
+    terminal::{Clear, ClearType},
+};
+use log::error;
+use rayon::ThreadPoolBuilder;
 use sha3::{Digest, Keccak256};
-use log::{error, warn};
-use std::time::Duration;
+use std::error::Error as StdError;
+use std::fmt;
+use std::io::{stdout, Write};
+use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::Arc;
+use std::thread;
+use std::time::{Duration, Instant};
+use tokio::sync::mpsc;
+
+// Add constant at the top of the file
+const CLIENT_NAME: &str = "cli_proof_node_v3";
+
+#[derive(Debug)]
```

Comment:
I feel like generally if you create a custom error we should try and capture some context via typing. We can do this easily with `thiserror` crate.

If we just want to propagate messages we can use the `anyhow` crate I mentioned before.

To return error you can use `bail!("Missing attribute: {}", missing);` macro.

It also allows you to easily wrap + rethrow errors:
```rs
some_errorable()..context("Failed to do the important thing")?;
```

Reply:
Added